### ISSUE TITLE
kube-dns addon fails in air gapped environment

### ISSUE THREAD

**Issue by linben** [mesosphere/dcos-kubernetes-quickstart#33]
the mandatory-addons-0-kube-dns task fails in an air gapped environment due to inability to pull the following resources

```
    "container": {
      "docker": {
        "kubedns": "gcr.io/google_containers/k8s-dns-kube-dns-amd64:1.14.5",
        "kubedns-dnsmasq": "gcr.io/google_containers/k8s-dns-dnsmasq-nanny-amd64:1.14.5",
        "kubedns-sidecar": "gcr.io/google_containers/k8s-dns-sidecar-amd64:1.14.5",
        "heapster": "gcr.io/google_containers/heapster-amd64:v1.4.2",
        "dashboard": "gcr.io/google_containers/kubernetes-dashboard-amd64:v1.6.3"
      }
    },
```

**Comment by ted-jung** [comment]
Hi, linben
I also have the same issue while met installing kubernetes on DC/OS(community edition)
following is my last error message that i got.
======================
 ######  Deploying kube-dns components --  ###### 
configmap "kube-dns" configured
service "kube-dns" configured
deployment "kube-dns" configured
kube-dns check failed and is not ready
================================

Rest of other containters come up well except mandatory-addon-0-kube-dns.
Do you still have the issues? or already resolved?
If you have resolved then, share your solution.

Thanks,
Ted

**Comment by linben** [comment]
Hi @ted-jung 

In addition to the above images, you need ```gcr.io/google_containers/pause-amd64:3.0```

Make those images available locally in your image registry. Then run ```kubectl replace -f <component.yaml>``` for kube-dns, heapster, and dashboard (after modifying the configs to point to your private registry images). 

Do ```kubectl get pods --all-namespaces``` to see status of these components 

The next release will incorporate these changes.

-ben

**Comment by hectorj2f** [comment]
@ted-jung @linben In our next release, we'll include a fix to tackle this issue. In any case, the registry from which to pull these docker images can be configured in our package via `resource.json` although we recommend to use the `gcr.io/google_containers` registry to avoid undesired issues.

In the following releases, we might include some flags to disable the deployment of heapster and dashboard.

**Comment by pires** [comment]
Should be fixed in 0.3.0-1.7.10-beta.
